### PR TITLE
Framework: create "blocks" directory for application components

### PR DESCRIPTION
**Pull request by mtias**
So far we have called these a bit confusingly app-components, but we haven't separated them under the folder structure. They are scattered around section folders (my-sites, post-editor, reader, signup, me) and the ui components folder.

It's time to have a dedicated place for them in our folder structure to help developers know what is available to them.

Devdocs has also been updated to the following:

image

----

The first block we are moving is `Site`, used throughout Calypso to render a site card. 

Test live: https://calypso.live/?branch=add/blocks

### CODE REVIEW

File: client/blocks/site/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ This component displays a Site item using the data structure provided by sites-l
 #### How to use:
 
 ```js
-var Site = require( 'my-sites/site' );
+var Site = require( 'blocks/site' );
```
Comment (ehg) [inline client/blocks/site/README.md]: ES6ify while we're here?

Comment (mtias) [inline client/blocks/site/README.md]: Would rather get this with the minimal possible change to avoid breaking anything. :)

**Comment by ehg** [comment]
Could we add a `README.md` explaining what blocks are?

File: client/blocks/site/README.md (hunk 1 above)
Comment (ehg) [inline client/blocks/site/README.md]: I doubt changing a `README.md` example is gonna break things ;) I get your point though.

Comment (mtias) [inline client/blocks/site/README.md]: oh, I thought you meant all the cases

**Comment by mtias** [comment]
@ehg good call, added an initial document. It can be expanded and improved.

**Comment by ehg** [comment]
Tests fine using e2e tests, which cover most of the changed files, except Jetpack.

**Comment by ehg** [comment]
Added 6548daee483a02ab6aa4493186095273aae33da5, which I guess you can squash into the devdocs change commit. 

After that, LGTM!

**Pull request merged**